src/main/java/uk/gov/hmcts/darts/common/service/impl/HearingCommonServiceImpl.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.hmcts.darts.common.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.darts.common.entity.CourtCaseEntity;
+import uk.gov.hmcts.darts.common.entity.CourtroomEntity;
+import uk.gov.hmcts.darts.common.entity.HearingEntity;
+import uk.gov.hmcts.darts.common.entity.MediaEntity;
+import uk.gov.hmcts.darts.common.entity.UserAccountEntity;
+import uk.gov.hmcts.darts.common.repository.HearingRepository;
+import uk.gov.hmcts.darts.common.service.CaseCommonService;
+import uk.gov.hmcts.darts.common.service.CourtroomCommonService;
+import uk.gov.hmcts.darts.common.service.HearingCommonService;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Service
+public class HearingCommonServiceImpl implements HearingCommonService {
+
+    private final HearingRepository hearingRepository;
+    private final CaseCommonService caseCommonService;
+    private final CourtroomCommonService courtroomCommonService;
+
+    @Override
+    @Transactional
+    public HearingEntity retrieveOrCreateHearing(String courthouseName, String courtroomName, String caseNumber, LocalDateTime hearingDate,
+                                                 UserAccountEntity userAccount) {
+        Optional<HearingEntity> foundHearing = hearingRepository.findHearing(courthouseName, courtroomName, caseNumber, hearingDate.toLocalDate());
+
+        return foundHearing.map(hearingEntity -> setHearingLastDateModifiedBy(hearingEntity, userAccount))
+            .orElseGet(() -> createHearing(courthouseName, courtroomName, caseNumber, hearingDate, userAccount, null));
+    }
+
+    @Override
+    @Transactional
+    public HearingEntity retrieveOrCreateHearingWithMedia(String courthouseName, String courtroomName, String caseNumber, LocalDateTime hearingDate,
```

Comment:
i was confused by this method, and so looked into its usages and realised ti seems to be some helper method for tests. it probably shouldn't be in here.
Can we add a comment in the code for it to be removed at a later date?

Reply:
sure added @Deprecated with detail